### DISCUSSION TITLE
optimal number of trees for random survival forest

### DISCUSSION THREAD

**Discussion by drpiment** [kogalur/randomForestSRC#356] [Q&A]
Hi,
We used with success the Random Survival Forest (RSF) in R to develop a molecular signature to predict disease recurrence. We started with 14 variables in a dataset of 400 patients and used the rfsrc function of the randomforestSRC package with the default number of trees (1000) along with the other default parameters. We repeated this 10 times to average the variable importance of each variables and selected those with positive values (6 variables). Those 6 variables were then used to create a cox model (cph funtion of the rms package).
One of the reviewers of our manuscript is asking for the motivation for selecting this number of trees regarding the model's efficiency. I am wondering what the best way would be to address this question. I haven’t found much literature addressing this specific topic in the context of RSF. Following this reviewer’s question, we tested using 800, 1000, 1200, 2000 and 4000 trees in RSF and always end up with the same 6 variables having positive variable importance values. I pasted below an example of error rate plot generated by using 4000 trees.
Thank you in advance for your guidance!
![image](https://user-images.githubusercontent.com/123601524/214708663-74aff340-d8bc-4185-b080-c9033879b608.png)

**Comment by ishwaran** [comment]
Perhaps you can show the confidence intervals for the variable importance? This would show the level of uncertainty and if the variables have importance that is statistically positive, then the confidence region would exclude zero.  The function [subsample](https://www.randomforestsrc.org/reference/subsample.rfsrc.html) (also see this [vignette](https://www.randomforestsrc.org/articles/rfsrc-subsample.html) will produce confidence regions for variable importance.  Examples are provided in the links I sent.

**Reply by drpiment** [reply]
Thank you! We will try that. Are there any guidelines/recommendations on how to determine the number of trees required to properly perform random survival forest?

**Comment by ishwaran** [comment]
The error plot you showed is the method for setting number of trees.  It shows the cumulative out-of-sample error as number of trees increases, and due to the law of large numbers converges to the estimated generalization error for the random forest model.  

Informally stated this means that once the plot begins to stabilize, you can be confident that adding more trees will not further improve the achievable error performance of the random forest model.  Therefore, there is no need to add more trees beyond this.
